L1Trigger/TrackFindingTMTT/src/L1KalmanComb.cc

```diff
@@ -0,0 +1,1828 @@
+
+          if (getSettings()->kalmanFillInternalHists())
+            fillStepHists(tpa, iteration, new_state);
+
+          // Cut on track chi2, pt etc.
+          if (isGoodState(*new_state))
+            next_states.push_back(new_state);
+        }
+
+        // loop over each stub in next layer if we skip, and check for compatibility with this state
+        for (unsigned i = 0; i < next_stubs.size(); i++) {
+          const StubCluster *next_stubCluster = next_stubs[i];
+
+          const KalmanState *new_state =
+              kalmanUpdate(skipped + 1 + nSkippedDeadLayers_nextStubs + nSkippedAmbiguousLayers_nextStubs,
+                           layer + 2 + nSkippedDeadLayers_nextStubs + nSkippedAmbiguousLayers_nextStubs,
+                           next_stubCluster,
+                           *the_state,
+                           tpa);
+
+          if (getSettings()->kalmanFillInternalHists())
+            fillStepHists(tpa, iteration, new_state);
+
+          if (isGoodState(*new_state))
+            next_states_skipped.push_back(new_state);
+        }
+
+        // post Kalman filter local sorting per state
+        sort(next_states.begin(), next_states.end(), KalmanState::orderChi2);
+        sort(next_states_skipped.begin(), next_states_skipped.end(), KalmanState::orderChi2);
+
+        int i, max_states, max_states_skip;
+
+        // If layer contained several stubs, so several states now exist, select only the best ones.
+        // -- Disable this by setting to large values, as not used in latest KF firmware.
+        // (But not too big as this wastes CPU).
+
+        switch (iteration) {
+          case 0:
+            max_states = 15;
+            max_states_skip = 15;
+            break;
+          case 1:
+            max_states = 15;
+            max_states_skip = 15;
+            break;
+          case 2:
+            max_states = 15;
+            max_states_skip = 15;
+            break;
+          case 3:
+            max_states = 15;
+            max_states_skip = 15;
+            break;
+          case 4:
+            max_states = 15;
+            max_states_skip = 15;
+            break;
+          case 5:
+            max_states = 15;
+            max_states_skip = 15;
+            break;
+          default:
+            max_states = 15;
+            max_states_skip = 15;
+            break;
+        }
+
+        i = 0;
+        for (auto state : next_states) {
+          if (i < max_states) {
+            new_states.push_back(state);
+          } else {
+            break;
+          }
+          i++;
+        }
+
+        i = 0;
+        for (auto state : next_states_skipped) {
+          if (i < max_states_skip) {
+            new_states.push_back(state);
+          } else {
+            break;
+          }
+          i++;
+        }
+
+      }  //end of state loop
+
+      if (getSettings()->kalmanFillInternalHists()) {
+        TString hname = Form("hstubComb_itr%d", iteration);
```

Comment:
```suggestion
        std::string hname = "hstubComb_itr"+std::to_string(iteration);
```

Reply:
The internal histograms have all been deleted from the Kalman Fit code.